### PR TITLE
Correct bug in vector handling due to fiona not correctly parsing Pat…

### LINKED ISSUE #198: Support zipped shapefiles in wps_generic_shape_properties
## Description

The current implementation chokes on a shapefile sent within a zipfile.

### PR DESCRIPTION
**Pull request by Zeitsperre**
…hlib objects

## Overview

This PR fixes #198 

Changes:

* Made it so that unzipped files from `generic_extract_archive` return lists of strings and not lists of PosixPath objects.

## Related Issue / Discussion
The issue with this has to do with a change implemented by the `fiona` maintainers that changed how certain objects were handled. This PR makes it so that unzipped or unarchived objects are returned as type `str` and not `PosixPath`.

## Additional Information

Links to other issues or sources.

https://github.com/Toblerity/Fiona/issues/836

### CODE REVIEW

**Comment by Zeitsperre** [comment]
@huard For some reason `test_ERA5::TestRavenERA5` is hanging. Just did a full reinstall of the most current Raven and both tests won't advance. I've marked them as skipped in this PR in order to advance on this.

**Comment by richardarsenault** [comment]
@Zeitsperre The hanging could be because the code collects data on the THREDDS server and it seems it was slower than usual today. Sometimes took me 1-2 minutes before, but today took me 5-10+.

Also basic question but how can I test this fix with code in another branch...?

**Comment by Zeitsperre** [comment]
@richardarsenault That would explain why I've had such a terrible time writing jupyter notebooks for xclim today. Good to know.

In order to test this, you could try rebasing your fork on this PR. I've only done that a handful of times and I'd encourage you to not push changes you make afterwards, but that would allow you to evaluate whether these changes will work with what you have.

**Comment by huard** [comment]
You `git pull` then `git merge fix_198` in your branch. 

There is a large data transfer on the thredds server, slowing everything down.

File: tests/conftest.py [hunk 1]
```diff
@@ -41,7 +42,7 @@ def params(ts_stats, tmp_path):
 def era5_hr():
     import netCDF4 as nc
     """Return a netCDF file with hourly ERA5 data at the Salmon location."""
-    path = TD / "ERA5" / "ts.nc"
+    path = Path(TD) / "ERA5" / "ts.nc"
```
Comment (huard) [inline tests/conftest.py]: TD should already be a Path instance.

File: tests/test_ERA5.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
           2.6851, 0.3740, 1.0000, 0.4739, 0.0114, 0.0243, 0.0069, 310.7211, 916.1947)
 
 
+@pytest.mark.skip(reason="Hanging test")
```
Comment (huard) [inline tests/test_ERA5.py]: ok for now. The NcML work is meant to keep data on the server and should speed this up.

File: tests/conftest.py (hunk 1 above)
Comment (Zeitsperre) [reply tests/conftest.py]: True. I wasn't sure why Path was imported. Given that TD is a class instance, I suppose there isn't a need for that library then.

File: tests/test_ERA5.py (hunk 2 above)
Comment (Zeitsperre) [reply tests/test_ERA5.py]: Keep it for now, or remove and merge this PR later?

**Comment by richardarsenault** [comment]
Sorry for beginner question, still learning: Just tried doing (from my REA5_Notebook branch):

git pull
git merge master (because the fix_198 has been merged into master already)
make start 
make notebook

And my code is still failing at the same place, i.e. the PosixPath for the shapefile is invalid. Any ideas? Can I merge my notebook in the master and restart a new branch to fix this?

**Comment by Zeitsperre** [comment]
Try git fetch and follow through with the rest of the merge process.

Doing just a git pull will update the branch that you're currently on. git fetch will update all existing branches.

**Comment by huard** [comment]
git fetch only updates the metadata. 
You need to do
```
git co master
git pull
git co ERA5_notebook
git merge master
```

**Comment by Zeitsperre** [comment]
Oof. I'm really out of practice. Good to know.

**Pull request merged**